### PR TITLE
CTCTRADERS-1231

### PR DESCRIPTION
**Pull request by amaalali**
- Remove Guice from spec base
- move app builder with nunjucks mock to it's own trait
- Make user answers generator lazy

### CODE REVIEW

**Review by ashwinirudrappa — APPROVED** [review]

File: test/base/MockNunjucksRendererApp.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+trait MockNunjucksRendererApp extends GuiceOneAppPerSuite with BeforeAndAfterEach with MockitoSugar {
+  self: TestSuite =>
+
+  val mockRenderer: NunjucksRenderer = mock[NunjucksRenderer]
+
+  override def beforeEach {
+    Mockito.reset(
+      mockRenderer
+    )
+    super.beforeEach()
```
Comment (alexflav23) [inline test/base/MockNunjucksRendererApp.scala:50]: You're resetting the execution order here. E.g could there be something that needs to run before your override or did you start the mockito reset before anything else on purpose?

**Review by alexflav23 — APPROVED** [review]

File: test/base/MockNunjucksRendererApp.scala (hunk 1 above)
Comment (amaalali) [reply test/base/MockNunjucksRendererApp.scala:50]: An arbitrary decision, do you think this should be done after call to super?

Comment (alexflav23) [reply test/base/MockNunjucksRendererApp.scala:50]: @amaalali I don't know if it's relevant tbh, just asked to be sure. I usually put it after only if there's some sequence of events that matters in the setup. If it doesn't, well then whatever.

Comment (amaalali) [reply test/base/MockNunjucksRendererApp.scala:50]: It looks like it doesn't matter at the moment but I'll keep an eye on it

**Pull request merged**